### PR TITLE
resolve file name with abs path but different case in drive label (closes #3186)

### LINKED ISSUE #3186: TestRun can't be resolved when an absolute path is used (VSCode integration, extensions)
TestRun can't be resolved when test runs the following way:
```bash
cd <absolute_path> && "<path_to_nodejs>\node.exe" node_modules\testcafe\lib\cli\index.js chrome <absolute_path>\tests\test.js
```
It's the common issue for integration (VS Code Test Runner etc.)

### What is your Test Scenario?
```js
import { waitForReact } from 'testcafe-react-selectors';

fixture('Bound test run')
	.page(`https://reactjs.org/`)
	.beforeEach(async () => {
	  await waitForReact();
	});

test('do smth', async t => {

});
```
 
### What is the Current behavior?
```
ClientFunction cannot implicitly resolve the test run in context of which it should be executed. If you need to call ClientFunction from the Node.js
      API callback, pass the test controller manually via ClientFunction's `.with({ boundTestRun: t })` method first. Note that you cannot execute
      ClientFunction outside the test code.
```

### PR DESCRIPTION
**Pull request by AlexKamaev**

### CODE REVIEW

File: src/utils/parse-file-list.js [hunk 1]
```diff
@@ -33,6 +46,9 @@ async function convertDirsToGlobs (fileList, baseDir) {
 
             if (fileStat.isDirectory())
                 return path.join(file, TEST_FILE_GLOB_PATTERN);
+
+            if (OS.win)
+                file = modifyFileRoot(baseDir, file);
```
Comment (miherlosev) [inline src/utils/parse-file-list.js]: It looks to complex.
I was be able to fix the problem using the following code:
```js
const absPath = path.isAbsolute(file) ? file : path.resolve(baseDir, file);
```

Comment (miherlosev) [reply src/utils/parse-file-list.js]: We can simplify `modifyFileRoot (baseDir, file)` as follows:

```js
function modifyFileRoot (baseDir, file) { 
    const absPath = path.isAbsolute(file) ? file : path.resolve(baseDir, file);

   return baseDir[0] + absPath.slice(1);
}
```

@kirovboris will we do this or stay it as is?

Comment (kirovboris) [reply src/utils/parse-file-list.js]: It looks like it doesn't work in the following case:
```bash
node C:\...\node_modules\testcafe\lib\cli\index.js chrome D:\...\test.js
```
We will try to run `C:\...\test.js` instead. But I agree the current implementation doesn't look very readable.

Comment (AndreyBelym) [reply src/utils/parse-file-list.js]: I doubt if we should compare only the drive letter in the case-insensitive way.  E.g. I can mistype and use an uppercase or lowercase letter in any path fragment on Windows, e.g.
```
.\node_modules\.bin\testcafe chrome D:\test\release\test.js
.\node_modules\.bin\testcafe chrome D:\Test\release\test.js
.\node_modules\.bin\testcafe chrome D:\TEST\release\test.js
```

will run the same `test.js` on Windows, but will break test run detection.

Comment (kirovboris) [reply src/utils/parse-file-list.js]: Sure, it makes sense, just the error with disk letter is more common.

**Pull request merged**